### PR TITLE
Add --ranks-per-node option to tester

### PR DESCRIPTION
**Pull request by bryevdv**
This PR adds support for `--node-per-rank` to `legate.tester`

### CODE REVIEW

**Review by manopapad — CHANGES_REQUESTED** [review]
Mostly nits, and what looks like one double-counting of `config.ranks` in GPU phase `compute_spec`.

File: legate/tester/stages/_linux/gpu.py [hunk 1]
```diff
@@ -56,30 +56,42 @@ def delay(self, shard: Shard, config: Config, system: TestSystem) -> None:
...
+            for j in range(config.ranks):
+                shard_gpus = range(
+                    (j + i * config.ranks) * config.gpus,
+                    (j + i * config.ranks + 1) * config.gpus,
+                )
+                shard = tuple(shard_gpus)
+                rank_shards.append(tuple(sorted(shard)))
+            shards.append(Shard(rank_shards))
+
+        return StageSpec(workers, shards * workers)
```
[the remark this answers is not in the document]
Comment (bryevdv) [reply legate/tester/stages/_linux/gpu.py:97]: This change results in two additional shards, just confirming that's what you what?
```
E         Left contains 2 more items, first extra item: Shard(ranks=[(0, 1), (2, 3)])
E         Full diff:
E           [
E            Shard(ranks=[(0, 1), (2, 3)]),
E            Shard(ranks=[(0, 1), (2, 3)]),
E         +  Shard(ranks=[(0, 1), (2, 3)]),
E         +  Shard(ranks=[(0, 1), (2, 3)]),...
```

[the remark this answers is not in the document]
Comment (manopapad) [reply legate/tester/stages/_linux/gpu.py:97]: I just tried a simple test, and I think this change, and the one from `int(degree * oversub_factor / config.ranks)` to `degree * oversub_factor`, is correct. Can you fact-check me?

I made a sample machine with 4 GPUs with 3GB framebuff each:

```
       return (GPUInfo(0, 3 * (1 << 30)),
               GPUInfo(1, 3 * (1 << 30)),
               GPUInfo(2, 3 * (1 << 30)),
               GPUInfo(3, 3 * (1 << 30)))
```

In this case, for ` --fbmem 1000 --gpus 1 --ranks-per-node 2` I think there should be 4 workers; two using GPUs 0 and 1 for their two ranks, and two using GPUs 2 and 3, given that each only uses 1.5GB (with bloat factor) of the available 3GB.

Here's what I see with the original code:

```
~/cunumeric> ./test.py --dry-run --use cuda --fbmem 1000 --gpus 1 --ranks-per-node 2
N = 4
degree = 2
fbsize = 3072.0
oversub_factor = 2
workers = 2
shards * oversub_factor = [Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)]), Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)])]
shards * workers = [Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)]), Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)])]
```

And here's what I get after replacing `int(degree * oversub_factor / config.ranks)` with `degree * oversub_factor`:

```
~/cunumeric> ./test.py --dry-run --use cuda --fbmem 1000 --gpus 1 --ranks-per-node 2
N = 4
degree = 2
fbsize = 3072.0
oversub_factor = 2
workers = 4
shards * oversub_factor = [Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)]), Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)])]
shards * workers = [Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)]), Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)]), Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)]), Shard(ranks=[(0,), (1,)]), Shard(ranks=[(2,), (3,)])]
```

So it appears that `shards * oversub_factor` is sufficient for the return value as well.

**Comment by bryevdv** [comment]
@manopapad I pushed changes for OMP let me know if this matches intent

File: legate/tester/stages/_linux/omp.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def shard_args(self, shard: Shard, config: Config) -> ArgList:
         if config.cpu_pin != "none":
             args += [
                 "--cpu-bind",
-                ",".join(str(x) for x in shard),
+                str(shard),
             ]
```
Comment (manopapad) [inline legate/tester/stages/_linux/omp.py:66]:
```suggestion
            ]
        if config.ranks > 1:
            args += [
                "--ranks-per-node",
                str(config.ranks),
            ]
```

Doesn't look like `--ranks-per-node` is added in this path.

**Review by manopapad — CHANGES_REQUESTED** [review]

**Review by manopapad — APPROVED** [review]
LGTM. The CI failures are unrelated, and can be ignored.

**Pull request merged**